.clang-tidy

```diff
@@ -10,7 +10,12 @@ Checks: '
     -cppcoreguidelines-pro-bounds-array-to-pointer-decay,
     readability-*,
     google-explicit-constructor,
-    -readability-identifier-length'
+    -readability-identifier-length,
+    -cppcoreguidelines-avoid-const-or-ref-data-members,
+    -bugprone-unchecked-optional-access,
+    -cppcoreguidelines-avoid-do-while,
+    -modernize-use-emplace,
+    -readability-convert-member-functions-to-static'
```

Comment:
I don't feel particularly strongly about removing any of these, but since some time has passed since this was PRed maybe it's worth reconsidering whether we remove these or not.

Unchecked optional access and modernize use emplace seem like two we might want to keep around

Reply:
It's been long enough since I opened this PR, that I don't remember my exact reasoning; but in general, I'd always want to avoid removing tidy plugins. If my hazy memory serves correctly, I removed these because they were flagging quite a lot of things in the codebase, so there'll be some churn when we enable them.

Not a problem at all, but should be done in a different PR.